base/util.go

```diff
@@ -1704,3 +1704,14 @@ func CoalesceBools(a, b *bool) *bool {
 	}
 	return nil
 }
+
+// CoalesceTimes returns the first non-nil argument, or nil if both are nil.
```

Comment:
I couldn't find a usage for this function in this PR?  Is it still useful/needed?

Reply:
Good catch - should have been used in `PrincipalConfig.Merge` but looks like it got accidentally removed in a rebase.

Reply:
(This actually exposed a gap in the test coverage, as it looks like `Merge` wasn't exercised. I'll add an extra test that should hit this path.)